app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,73 @@
+class DriversController < ApplicationController
+  def index
+    @drivers = Driver.all
+  end
+
+  def show
+    driver_id = params[:id].to_i
+    @driver = Driver.find_by(id: driver_id)
+
+    if @driver.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def edit
+    @driver = Driver.find_by(id: params[:id])
+
+    if @driver.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @driver = Driver.find_by(id: params[:id])
+
+    if @driver.nil?
+      head :not_found
+      return
+    elsif @driver.update(driver_params)
+      redirect_to driver_path(@driver.id)
+      return
+    else
+      render :edit
```

Comment:
Just a suggestion if the validations failed to show the proper response code

```suggestion
      render :edit, status: :bad_request
```